test/icons.test.ts

```diff
@@ -18,6 +18,7 @@ describe('getNodeIcon', () => {
     });
 
     test('handles invalid input', () => {
+        // @ts-expect-error The function expects strings, but test an invalid input just to be sure
```

Comment:
Not sure about this one - why would we want to 'test an invalid input'? I guess we should either remove or fix this test, or align function declaration if it in fact supports number arguments. Applies to L50 as well.

Reply:
I see your point. We can ditch the _safeguards_ that were useful before. With TypeScript, we can be _almost_ sure that the function will get correct input.

However, TypeScript still allows using the `any` type, e.g.:
```ts
const arg = null as any;
getNodeIcon(arg); // no error here, `any` is assignable to `string`
```

So, all in all, we:
1. Suggest that the function accepts only strings, via TypeScript. This ensures that consumers know what to pass it
2. In the case that a consumer passes an invalid value, we still guard against it.

I would rather not change the function implementation to avoid breaking changes, and this test still is useful to me, since it tests that the function also works fine with unexpected values (meaning the test verifies the existing behavior)

Reply:
If the function works correctly with `number` argument, then lets just extend its TS declaration - no change in implementation is required, right?
Or, if the `number` type doesn't make sense due to function semantics, then I don't think it does matter how the function behaves for such argument. In fact, passing a `number` argument would probably mean a bug, so if I expected something in such scenario then it would be a thrown exception.

Reply:
I think it's the latter scenario - only strings seem to be valid arguments. I don't want to make breaking changes in this PR and possibly cause some runtime errors if this function was used incorrectly :smile: What I can do is remove the test. Would that suffice?

Reply:
Sure, I'm fine with it

Reply:
Done in 30a5d39046e652396bd124e55330d89035d77305